src/ORM/Association/HasMany.php

```diff
@@ -182,6 +182,174 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     }
 
     /**
+     * Associates the source entity to each of the target entities provided.
+     * When using this method, all entities in `$targetEntities` will be appended to
+     * the source entity's property corresponding to this association object.
+     *
+     * This method does not check link uniqueness.
+     *
+     * ### Example:
+     *
+     * ```
+     * $user = $users->get(1);
+     * $allArticles = $articles->find('all')->execute();
+     * $users->Articles->link($user, $allArticles);
+     * ```
+     *
+     * `$user->get('articles')` will contain all articles in `$allArticles` after linking
+     *
+     * @param \Cake\Datasource\EntityInterface $sourceEntity the row belonging to the `source` side
+     * of this association
+     * @param array $targetEntities list of entities belonging to the `target` side
+     * of this association
+     * @param array $options list of options to be passed to the save method
+     * @return bool true on success, false otherwise
+     */
+    public function link(EntityInterface $sourceEntity, array $targetEntities, array $options = [])
+    {
+        $saveStrategy = $this->saveStrategy();
+        $this->saveStrategy(self::SAVE_APPEND);
+        $property = $this->property();
+
+        $currentEntities = (new Collection((array)$sourceEntity->get($property)))->append($targetEntities);
```

Comment:
I thought I would be more direct and readable to do that, but I can switch to pure arrays.